internal/osbuildexecutor/runner-impl-aws-ec2_test.go

```diff
@@ -0,0 +1,194 @@
+package osbuildexecutor_test
+
+import (
+	"archive/tar"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/osbuild/images/pkg/osbuild"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/osbuild/osbuild-composer/internal/osbuildexecutor"
+)
+
+func TestWaitForSI(t *testing.T) {
+	server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*1)
+	defer cancel()
+	require.False(t, osbuildexecutor.WaitForSI(ctx, server.URL))
+
+	server.Start()
+	ctx2, cancel2 := context.WithTimeout(context.Background(), time.Second*1)
+	defer cancel2()
+	require.True(t, osbuildexecutor.WaitForSI(ctx2, server.URL))
+}
+
+func TestWriteInputArchive(t *testing.T) {
+	cacheDir := t.TempDir()
+	storeDir := filepath.Join(cacheDir, "store")
+	require.NoError(t, os.Mkdir(storeDir, 0755))
+	storeSubDir := filepath.Join(storeDir, "subdir")
+	require.NoError(t, os.Mkdir(storeSubDir, 0755))
+
+	require.NoError(t, os.WriteFile(filepath.Join(storeDir, "contents"), []byte("storedata"), 0600))
+	require.NoError(t, os.WriteFile(filepath.Join(storeSubDir, "contents"), []byte("storedata"), 0600))
+
+	archive, err := osbuildexecutor.WriteInputArchive(cacheDir, storeDir, []string{"image"}, []byte("{\"version\": 2}"))
+	require.NoError(t, err)
+
+	cmd := exec.Command("tar",
+		"-tf",
+		archive,
+	)
+	out, err := cmd.CombinedOutput()
+	require.NoError(t, err)
+	require.ElementsMatch(t, []string{
+		"control.json",
+		"manifest.json",
+		"store/",
+		"store/subdir/",
+		"store/subdir/contents",
+		"store/contents",
+		"",
+	}, strings.Split(string(out), "\n"))
+}
+
+func TestHandleBuild(t *testing.T) {
+	buildServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusCreated)
```

Comment:
Maybe for a followup - it might be nice to read the input there and doulbe check that we got the expected inputArchive, this might be as simple as `inp := io.ReadAll(r.Body); c.AssertEqual(t, []byte("test"), inp)` or something.

Reply:
https://github.com/osbuild/osbuild-composer/pull/4204